crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -115,7 +119,42 @@ impl Trade {
                     },
                 }
             }
-            order::Kind::Liquidity => todo!(),
+            order::Kind::Liquidity => {
+                // Liquidity orders (including JIT) compute the executed amounts by linearly
+                // scaling the buy/sell amounts in the order.
```

Comment:
Theoretically the solver could return an `executed_amount` smaller than the total amount for a fill-or-kill order. Not sure if this needs to be caught in the asset flow test or if it's enough to find the error with the failing simulation.
I'm not requesting a change for this PR. Just wanted to bring it up in case we already want to detect that case before simulating.